app/controllers/dashboards_controller.rb

```diff
@@ -1,4 +1,30 @@
 # frozen_string_literal: true
 class DashboardsController < ApplicationController
-  def index; end
+  include Flashable
+
+  def index
+    @uploads = Upload.last_uploads
+  end
+
+  def build
+    @version = InstitutionBuilder.run(current_user)
+  end
+
+  def export
+    klass = csv_model(params[:csv_type])
+    csv_data = klass.export
+    raise StandardError, 'No data to export.' if csv_data.blank?
+
+    respond_to do |format|
+      format.csv { send_data csv_data, type: 'text/csv' }
+    end
+  rescue StandardError => e
+    redirect_to dashboards_path, alert: e.message
+  end
```

Comment:
So I'm not crazy about this pattern. I would have preferred using accepts nested attributes somehow, but I think that will have to happen in a future refactor. The reason being it is a generic controller action, not resourceful.